js/src/components/adaptive-form/adaptive-form.js

```diff
@@ -56,9 +63,35 @@ function AdaptiveForm( { children, ...props }, ref ) {
 		...formRef.current,
 	} ) );
 
+	// Enhancements related.
+	// Additional enhancements to make AdaptiveForm have more useful or reusable features.
+	// It could also be the playground of the practical instances before pushing them upstream.
+	const isMounted = useIsMounted();
+	const [ submission, setSubmission ] = useState( [ false, false ] );
+	const [ isSubmitting, isSubmitted ] = submission;
+
+	if ( onSubmit ) {
```

Comment:
❓ 
In which circumstances we don't need a submit function in the AdaptiveForm?

Reply:
(Asking this because is marked as an optional and also we have a check here for it's presence)

Reply:
It is optional in order to keep it consistent with the existing interface of WC's `Form` component. Ref:
- https://github.com/woocommerce/woocommerce/blob/6.9.0/packages/js/components/src/form/form.tsx#L43
- https://github.com/woocommerce/woocommerce/blob/6.9.0/packages/js/components/src/form/form.tsx#L82
- https://github.com/woocommerce/woocommerce/blob/6.9.0/packages/js/components/src/form/form.tsx#L229-L231